code/game/machinery/Sleeper.dm

```diff
@@ -383,12 +382,20 @@
 	if(panel_open)
 		to_chat(user, "<span class='notice'>Close the maintenance panel first.</span>")
 		return
-	if(dir == EAST)
-		orient = "LEFT"
-		setDir(WEST)
-	else
-		orient = "RIGHT"
-		setDir(EAST)
+
+	switch(dir)
+		if(EAST)
+			setDir(SOUTH)
+			return
+		if(SOUTH)
+			setDir(WEST)
+			return
+		if(WEST)
+			setDir(NORTH)
+			return
+		else
+			setDir(EAST)
+			return
```

Comment:
The entire switch block can be condensed into:
```dm
setDir(turn(dir, -90))
```
This sets the dir to the current dir 90 rotated degrees clockwise.